test/model/src/driver/kv_workload.cpp

```diff
@@ -37,6 +37,168 @@ extern "C" {
 
 namespace model {
 
+namespace operation {
+
+/*
+ * parse --
+ *     Parse an operation from a string. Throw an exception on error.
+ */
+any
+parse(const char *str)
+{
+    const char *p = str;
+
+    /* Get the operation name. */
+    std::string name;
+    while (*p != '\0' && (std::isalnum(*p) || *p == '_'))
+        name += *(p++);
+
+    while (*p != '\0' && std::isspace(*p))
+        p++;
+    if (*(p++) != '(')
```

Comment:
- Note, in C++ ++variable is more efficient than variable++ especially for user defined types since variable++ has to make a copy of the variable and return the copy.  Best to get in the habit of using ++variable unless you really need variable++.
- Note, std::isspace() returns false for '\0' so there is no need to test for '\0'. -- https://en.cppreference.com/w/cpp/string/byte/isspace

Bug: Line 58 *(p++) is one character past *p on line 56 since line 58 increments p before dereferencing it. I think line 58 is supposed to check whether the character on line 56 that was not a space is also not a '('

Thus change to the correct and more efficient:
while (std::isspace(*p))
      ++p;
 if (*p++ != '(')
     throw model_exception("Expected '('");

Reply:
The performance difference is primarily in user defined types, not integers. According to definition ++p increments first and then returns the modified value. p++ saves a copy of the original value, increments, and then returns the copy of the original value. That might be very slow with a user defined type because copying the user defined type could be very slow. I suggest using ++p all the time to get in the habit of it so the slow p++ is not done on user defined types unless that is really needed.

Reply:
Good to know! Now it makes sense. Thank you for explaining it.